### ISSUE TITLE
`env` context not available for `jobs.<job_id>.if`

### ISSUE THREAD

**Issue by thecristen** [rhysd/actionlint#155]
As the [context availability documentation indicates](https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability), using `env` isn't supported for a job's `if` condition. However this wasn't caught as an error by actionlint. In my case I was using a reusable workflow, so the job looked something like this:

```
jobs:
  pr:
    if: ${{ env.DO_DEPLOY }}
    uses: ./.github/workflows/use-deploy-ecs.yml
    with:
      # args here
    secrets:
      # blah blah blah and so on
```
